File: dotnet/cafdemalihapa/Files/Executables/IdeExecutiveFileLocation.cs
using Microsoft.Extensions.Logging;

namespace cafdemalihapa.Files.Executables
{
    public class IdeExecutiveFileLocation(
        ILogger<IdeExecutiveFileLocation> logger
    )
    {
        public string GetPath()
        {
            var ideExecuteFileLocation = CommandLineArgs.GetByKey("--ide-execute-file-location");
            logger.LogInformation("ide execute file location: {ideExecuteFileLocation}", ideExecuteFileLocation);
            return ideExecuteFileLocation;
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Files/Executables/IdeExecutiveFileLocation.fs
namespace cafdemwimu.console.Files.Executables

open Microsoft.Extensions.Logging
open cafdemwimu.console

type IdeExecutiveFileLocation(logger: ILogger<IdeExecutiveFileLocation>) =
    member _.GetPath() =
        let ideExecuteFileLocation = CommandLineArgs.GetByKey("--ide-execute-file-location")
        logger.LogInformation("ide execute file location: {ideExecuteFileLocation}", ideExecuteFileLocation)
        ideExecuteFileLocation
